services/QuillLMS/client/app/bundles/Shared/quill-marking-logic/src/libs/matchers/incorrect_sequence_match.ts

```diff
@@ -11,12 +11,19 @@ export function incorrectSequenceMatchHelper(responseString:string, incorrectSeq
 }
 
 export function incorrectSequenceMatch(responseString: string, incorrectSequences:Array<IncorrectSequence>):IncorrectSequence {
-  return _.find(incorrectSequences, (incSeq) => {
+  const sortedIncorrectSequences = incorrectSequences.sort((a, b) => (a.order || 0) - (b.order || 0));
+
+  for (let i = 0; i < sortedIncorrectSequences.length; i++) {
```

Comment:
You can still use underscore's `_.find` here -- [under the hood](https://underscorejs.org/docs/modules/find.html) it is also just doing a `for` loop, so it should always be iterating in order.